Sidebar: Update notices to look more like notices

shaunandrews opened the pull request:
Currently, we display notices in a different way in the sidebar. They don't have rounded borders, and when they stack, they tend to "meld" together.

<img width="311" alt="screen shot 2018-02-28 at 12 08 46 pm" src="https://user-images.githubusercontent.com/191598/36801733-df80b298-1c80-11e8-9394-2c64c378e30f.png">

This PR aims to make sidebar notices look more standard with rounded borders and adds a 4px margin:

<img width="299" alt="screen shot 2018-02-28 at 12 08 40 pm" src="https://user-images.githubusercontent.com/191598/36801776-fb4dafda-1c80-11e8-870b-b36d4b334920.png">

<img width="294" alt="screen shot 2018-03-01 at 2 22 31 pm" src="https://user-images.githubusercontent.com/191598/36865080-7782b448-1d5c-11e8-9d86-d514d919ba65.png">

<img width="290" alt="screen shot 2018-03-01 at 2 22 37 pm" src="https://user-images.githubusercontent.com/191598/36865081-7969a0fa-1d5c-11e8-9b27-d606528c5106.png">

drw158 commented:
I'm in favor of this change. I took some screenshots of some notices, not sure if the differences are intentional.

![download-10](https://user-images.githubusercontent.com/618551/36803214-8e1f4de6-1c7c-11e8-82c5-3c36264e37c1.png)
![download-7](https://user-images.githubusercontent.com/618551/36803215-8e2dd0fa-1c7c-11e8-819f-0f2d19e4de65.png)
![download-6](https://user-images.githubusercontent.com/618551/36803216-8e44c706-1c7c-11e8-9bf9-c9b5ad2df70c.png)
![screen shot 2018-02-28 at 11 27 29 am](https://user-images.githubusercontent.com/618551/36803217-8e5215c8-1c7c-11e8-9528-ff4c76b8df67.png)

shaunandrews commented:
@drw158 The green "notice-type" thing is called the SidebarBanner, and there's another PR thats looking to test some changes to that: https://github.com/Automattic/wp-calypso/pull/22755

shaunandrews commented:
Rebased. Moved the order of notices so the SidebarBanner (which @adamziel is testing in #22755) is always at the top, above the other notices.

<img width="290" alt="screen shot 2018-03-01 at 2 22 37 pm" src="https://user-images.githubusercontent.com/191598/36865137-a36acc26-1d5c-11e8-8835-3cd63f5e23a7.png">

aidvu commented on client/my-sites/current-site/notice.jsx:
```diff
@@ -138,11 +138,11 @@ class SiteNotice extends React.Component {
 				{ this.domainCreditNotice() }
 				{ this.jetpackPluginsSetupNotice() }
-				{ this.freeToPaidPlanNotice() }
 				<DomainToPaidPlanNotice />
```
Should also move this one, as it uses `SidebarBanner`.

shaunandrews replied:
Ah, didn't notice. I'll move it.

aidvu commented:
Pushed a small change, as the `DomainWarnings` were shown on top. @Automattic/cobalt how do you feel moving the notices below.

I hacked the code a bit to get these examples. Doubt there are more than 1-2 notices shown for a site.
**Before:**
![domains on top](https://user-images.githubusercontent.com/1103398/36938795-4ea6c83a-1f27-11e8-9de2-a4ced13507c4.png)
**After:**
![domains below](https://user-images.githubusercontent.com/1103398/36938802-564e1b92-1f27-11e8-81ba-a7ac3fe9e885.png)

shaunandrews commented:
Ah, perfect @aidvu — I figured there were more notices, but I couldn't find a way to trigger them all. Thanks for the updates.

klimeryk commented:
Thanks for proposing the reversed order of the notices there, @shaunandrews 😍 (upgrade nudges first, warnings second) They look so much better now - loving it 👍
